src/components/Scrollbar/Scrollbar.tsx

```diff
@@ -20,8 +21,52 @@ export const Scrollbar = ({
   ...props
 }: ScrollbarProps) => {
   const [ref] = useMeasure({ polyfill: ResizeObserver })
+  const timerRef = useRef<NodeJS.Timeout | null>(null)
+  const id = useMemo(() => uniqueId('scrollbar-'), [])
+
+  useEffect(() => {
+    return () => {
+      if (timerRef.current !== null) {
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        clearTimeout(timerRef.current)
+      }
+    }
+  }, [])
+
+  const hideScrollbar = () => {
+    const element = document.getElementById(id)
+    if (element) {
+      element.classList.remove('scrollbar--hovered-visible')
```

Comment:
For some reason the transition wouldn't work when I was trying to do this using classnames + state :/. Tried lots of different workarounds, but this was the simplest and most consistent.

Reply:
maybe a selector specificity thing? idk, this seems fine though

Reply:
yeah it was super weird

Reply:
like i did effectively the same thing using classnames + state but it didn't work